app/models/spree/gateway/amazon.rb

```diff
@@ -86,11 +92,16 @@ def capture(amount, amazon_checkout, gateway_options={})
       if amount < 0
         return credit(amount.abs, nil, nil, gateway_options)
       end
-      order = Spree::Order.find_by(:number => gateway_options[:order_id].split("-")[0])
+      order_number, payment_number = gateway_options[:order_id].split("-", 2)
```

Comment:
Can we extract the `gateway_options[:order_id].split("-", 2)` into a separate method to DRY it up and also maybe make it more verbose, maybe something like `extract_order_and_payment_number(gateway_options)`?